reqs/requirements-dev-snowflake.txt

```diff
@@ -1,2 +1,4 @@
 snowflake-connector-python>=2.5.0
 snowflake-sqlalchemy>=1.2.3
+# snowflake-sqlalchemy currently requires sqlalchemy<2.0.0,>=1.4.0
+sqlalchemy<2.0.0
```

Comment:
This is the "cleanest" way for the sqlachemy pin to live in `snowflake` world, but my concern is that the most up-to-date `sqlalchemy` will not be installed in cases where `dev` is requested. How concerned should I be about this?

Reply:
One idea would be to have the `pandas<2.0.0` pin and `sqlalchemy<2.0.0` pin in their own req files:

`[snowflake, sqlalchemy-less-2]`

and when all the connectors are eventualy compatible with sqlalchemy , then we can just change it to `[snowflake, sqlalchemy-less-2`]

Reply:
update: made the change.